ndk/src/main/java/com/bugsnag/android/ndk/NativeBridge.java

```diff
@@ -0,0 +1,413 @@
+package com.bugsnag.android.ndk;
+
+import com.bugsnag.android.Breadcrumb;
+import com.bugsnag.android.MetaData;
+import com.bugsnag.android.NativeInterface;
+
+import android.os.Build;
+import android.support.annotation.Nullable;
+import android.util.Log;
+
+import java.io.File;
+import java.util.List;
+import java.util.Observable;
+import java.util.Observer;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * Observes changes in the Bugsnag environment, propagating them to the native layer
+ */
+public class NativeBridge implements Observer {
+    static {
+        System.loadLibrary("bugsnag-ndk");
+    }
+
+    private static final int METADATA_SECTION = 0;
+    private static final int METADATA_KEY = 1;
+    private static final int METADATA_VALUE = 2;
+    private static final String LOG_TAG = "BugsnagNDK:NativeBridge";
+    private static final Lock lock = new ReentrantLock();
```

Comment:
Because this is static it is locking on the class rather than an instance of the class - is there any possibility NativeBridge could be instantiated more than once?

Reply:
Not really, no. It was largely insurance against a future change not causing a catastrophic result, as multiple native crash reporters could be.